codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/XmlShapeDeserVisitor.java

```diff
@@ -162,6 +162,13 @@ void deserializeNamedMember(
             sourceBuilder.append("['").append(targetLocation).append("']");
         }
 
+        // Handle self-closed xml parsed as an empty string.
+        if (deserializationReturnsArray) {
+            writer.openBlock("if ($L.$L === \"\") {", "}", inputLocation, locationName, () -> {
```

Comment:
Can you add some samples to the description to show how this handles unflattened aggregates?

Reply:
Updated description.